Neos.Neos/Classes/View/FusionView.php

```diff
@@ -99,161 +47,134 @@ public function render(): string|ResponseInterface
 
-    protected ?Runtime $fusionRuntime;
+    #[Flow\Inject]
+    protected FusionService $fusionService;
 
-    /**
-     * @Flow\Inject
-     * @var Context
-     */
-    protected $securityContext;
+    #[Flow\Inject]
+    protected Context $securityContext;
 
-    /**
-     * @param string $output
-     * @return string|ResponseInterface If output is a string with a HTTP preamble a ResponseInterface
-     *                                  otherwise the original output.
-     */
-    protected function parsePotentialRawHttpResponse($output)
+    #[Flow\Inject]
+    protected ContentRepositoryRegistry $contentRepositoryRegistry;
+
+    protected Runtime $fusionRuntime;
+
+    /** The Fusion path to use for rendering the node given in "value", defaults to "root". */
+    protected string $fusionPath = 'root';
+
+    protected FusionRenderingStuff $fusionRenderingStuff;
+
+    /** Set the Fusion path to use for rendering the node given in "value" */
+    public function setFusionPath(string $fusionPath): void
     {
-        if ($this->isRawHttpResponse($output)) {
-            return Message::parseResponse($output);
-        }
+        $this->fusionPath = $fusionPath;
+    }
 
-        return $output;
+    public function getFusionPath(): string
+    {
+        return $this->fusionPath;
     }
 
     /**
-     * Checks if the mixed input looks like a raw HTTTP response.
+     * Assigns the Node to this view. The key must be "value".
+     * The documentNode and siteNode as well the site are derived from this entry Node.
      *
-     * @param mixed $value
-     * @return bool
+     * @throws \Exception If 'value' is not a Node
      */
-    protected function isRawHttpResponse($value): bool
+    public function assign($key, $value): AbstractView
     {
-        if (is_string($value) && strpos($value, 'HTTP/') === 0) {
-            return true;
+        if ($key !== 'value') {
+            // noop -> we only allow "value"
+            return $this;
         }
 
-        return false;
-    }
+        // 'value' is expected to be a Node!
+        if (!$value instanceof Node) {
+            throw new Exception('FusionView needs a variable \'value\' set with a Node object.', 1329736456);
+        }
 
-    /**
-     * Is it possible to render $node with $his->fusionPath?
-     *
-     * @return boolean true if $node can be rendered at fusionPath
-     *
-     * @throws Exception
-     */
-    public function canRenderWithNodeAndPath()
-    {
-        $currentSiteNode = $this->getCurrentSiteNode();
-        $fusionRuntime = $this->getFusionRuntime($currentSiteNode);
+        // clear the cached runtime instance
+        unset($this->fusionRuntime);
 
-        return $fusionRuntime->canRender($this->fusionPath);
+        $this->fusionRenderingStuff = $this->renderingUtility->createFusionRenderingStuff(
+            $value,
+            $this->controllerContext->getRequest()
+        );
+        return $this;
```

Comment:
The biggest difference user wise is, that "value" will be eagerly resolved to "node" "documentNode" and etc. 

This is a mini performance gain, so that `canRenderWithNodeAndPath` and `render` dont have to each recompute the values.

This change is only noticeable when "value" is not a valid node. As exceptions are now thrown already in the "assign" step and not before hand.